### PR TITLE
docs: use nightly clippy in vscode

### PR DESCRIPTION
**Pull request by leruaa**
Before this small fix, I had a bunch of Clippy warnings/errors on VS Code.

### CODE REVIEW

**Comment by leruaa** [comment]
Wait, in fact, it's not working well

**Review by mattsse — COMMENTED** [review]
I'm not a RA user so can't really comment on this

wdyt @rkrasiuk

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -90,6 +90,14 @@ If you are working in VSCode, we recommend you install the [rust-analyzer](https
...
 "editor.formatOnSave": true,
 "rust-analyzer.rustfmt.extraArgs": ["+nightly"],
+"rust-analyzer.checkOnSave.overrideCommand": [
+  "cargo",
+  "+nightly",
+  "clippy",
+  "--all",
+  "--all-features",
+  "--message-format=json"
+],
```
Comment (onbjerg) [inline CONTRIBUTING.md:100]:
```suggestion
"rust-analyzer.checkOnSave.extraArgs": ["+nightly"],
```

**Review by onbjerg — CHANGES_REQUESTED** [review]
You can just add `+nightly`

File: CONTRIBUTING.md (hunk 1 above)
Comment (leruaa) [reply CONTRIBUTING.md:100]: Hmmm I have a similar error:
```
[ERROR flycheck] Flycheck failed to run the following command: "cargo" "clippy" "--workspace" "--message-format=json" "--manifest-path" "/workspaces/crypto/learn/ethereum/reth/reth/Cargo.toml" "--all-targets" "--all-features" "+nightly"
[ERROR rust_analyzer::lsp_utils] cargo check failed

Cargo watcher failed, the command produced no valid metadata (exit code: ExitStatus(unix_wait_status(256))):
error: Found argument '+nightly' which wasn't expected, or isn't valid in this context
```

Comment (onbjerg) [reply CONTRIBUTING.md:100]: Alright, strange. This is fine then, thank you for being patient!

**Review by onbjerg — APPROVED** [review]

**Pull request merged**